packages/flutter_test/lib/src/controller.dart

```diff
@@ -879,6 +923,27 @@ class LiveWidgetController extends WidgetController {
     await binding.endOfFrame;
   }
 
+  @override
+  Future<int> pumpAndSettle([
+    Duration duration = const Duration(milliseconds: 100),
+    Duration timeout = const Duration(minutes: 10),
+  ]) {
+    assert(duration != null);
+    assert(duration > Duration.zero);
+    assert(timeout != null);
+    assert(timeout > Duration.zero);
+    int count = 0;
+    return TestAsyncUtils.guard<void>(() async {
+      final DateTime endTime = clock.fromNowBy(timeout);
+      do {
+        if (clock.now().isAfter(endTime))
+          throw FlutterError('pumpAndSettle timed out');
+        await pump(duration);
+        count += 1;
+      } while (binding.hasScheduledFrame);
+    }).then<int>((_) => count);
```

Comment:
iirc you can make `guard` take an `int` and then just return `count` from the body.

Reply:
Done.